eZ/Publish/API/Repository/Repository.php

```diff
@@ -82,6 +82,29 @@ public function hasAccess($module, $function, UserReference $user = null);
      */
     public function canUser($module, $function, ValueObject $object, $targets = null);
 
+    /**
+     * Allows API execution to be performed with full access, sand-boxed.
+     *
+     * The closure sandbox will do a catch all on exceptions and rethrow after
+     * re-setting the sudo flag.
+     *
+     * Example use:
+     *     $location = $repository->sudo(function (Repository $repo) use ($locationId) {
+     *             return $repo->getLocationService()->loadLocation($locationId)
+     *         }
+     *     );
+     *
+     *
+     * @param callable $callback
+     * @param \eZ\Publish\API\Repository\Repository|null $outerRepository Optional, mostly for internal use but allows to
+     *                                                   specify Repository to pass to closure.
+     *
+     * @throws \Exception Re-throws exceptions thrown inside $callback
+     *
+     * @return mixed
+     */
+    public function sudo(callable $callback, self $outerRepository = null);
```

Comment:
I'm all for exposing sudo in API, but this belongs to `PermissionResolver`. Ideally we shouldn't require outer repository also, but if this cannot be solved right now, I can live with it ;)

Reply:
I can expose it there also, but as you say, it will requiore outerRepository, so it won't be as user friendly.

This is also why I for now opted to skip it, as it might be confusing to have same method with different requirements two places. WDYT?